compiler/rustc_mir_transform/src/elaborate_box_derefs.rs

```diff
@@ -91,8 +91,21 @@ impl<'tcx> MirPass<'tcx> for ElaborateBoxDerefs {
             let mut visitor =
                 ElaborateBoxDerefVistor { tcx, unique_did, nonnull_did, local_decls, patch };
 
-            for (bb, data) in basic_blocks.iter_enumerated_mut() {
-                visitor.visit_basic_block_data(bb, data);
+            for (block, BasicBlockData { statements, terminator, .. }) in
+                basic_blocks.iter_enumerated_mut()
+            {
+                let mut index = 0;
+                for statement in statements {
+                    let location = Location { block, statement_index: index };
+                    visitor.visit_statement(statement, location);
+                    index += 1;
+                }
+
+                // FIXME: horrible hack
```

Comment:
Oh so the problem was yield terminators? Why were they dereferencing boxes at all?

Reply:
You can do a `box yield` which puts the yielded value into the box in place.